### PR TITLE
Align interfaces of GenericRBReductor and GenericPGReductor

### LINKED ISSUE #473: Make GenericRBReductor and GenericPGReductor more coherent
`GenericPGReductor` is introduced in #464. In particular, we should look at the `product` and `biorthogonal_product` arguments.

### PR DESCRIPTION
**Pull request by sdrave**
Will fix #473.

### CODE REVIEW

**Comment by sdrave** [comment]
@pmli, I guess I am done with refactoring. Please take a careful look:

- I decided to remove all basis extension code from `GenericPGReductor` for now.
- I removed the inheritance of the sys-mor reductors from `GenericPGReductor` as things got quite messy with not calling `GenericPGReductor.__init__()`. Hope that's ok.

File: src/pymor/reductors/h2.py [hunk 1]
```diff
@@ -288,8 +289,8 @@ def reduce(self, rd0, tol=1e-4, maxit=100, num_prev=1, projection='orth', conv_c
         assert 0 < r < d.n
         assert isinstance(num_prev, int) and num_prev >= 1
         assert projection in ('orth', 'biorth')
-        self.biorthogonal_product = None if projection == 'orth' else 'E'
         assert conv_crit in ('sigma', 'h2')
+        self.pg_reductor = GenericPGReductor(d, self.W, self.V, projection == 'biorth', product=d.E)
```
Comment (pmli) [inline src/pymor/reductors/h2.py]: `self.W` and `self.V` are undefined here.

File: src/pymor/reductors/h2.py [hunk 2]
```diff
@@ -355,8 +356,9 @@ def _projection_matrices(self, rd, projection):
         elif projection == 'biorth':
             self.V, self.W = gram_schmidt_biorth(self.V, self.W, product=d.E)
```
Comment (pmli) [inline src/pymor/reductors/h2.py]: Maybe moving the line `self.pg_reductor = ...` from above to here would work?

Comment (pmli) [reply src/pymor/reductors/h2.py]: Just tried it, and yes, it would.

**Pull request merged**